internal/core/src/query/visitors/ExecExprVisitor.cpp

```diff
@@ -461,12 +461,35 @@ auto
 ExecExprVisitor::ExecTermVisitorImpl(TermExpr& expr_raw) -> BitsetType {
     auto& expr = static_cast<TermExprImpl<T>&>(expr_raw);
     auto& schema = segment_.get_schema();
-
+    auto primary_offset = schema.get_primary_key_offset();
     auto field_offset = expr_raw.field_offset_;
     auto& field_meta = schema[field_offset];
+
+    bool use_pk_index = false;
+    if (primary_offset.has_value()) {
+        use_pk_index = primary_offset.value() == field_offset && field_meta.get_data_type() == engine::DataType::INT64;
+    }
+
+    if (use_pk_index) {
+        auto id_array = std::make_unique<IdArray>();
+        auto dst_ids = id_array->mutable_int_id();
+        for (const auto& id : expr.terms_) {
+            dst_ids->add_data(id);
```

Comment:
expr.terms_ is a vector, here can use batch mode add

Reply:
how ?